tests/utils/managed_deployment.py

```diff
@@ -796,6 +828,20 @@ async def __aenter__(self):
             self._deployment_name = self.deployment_spec.name
             logging.getLogger("httpx").setLevel(logging.WARNING)
             await self._init_kubernetes()
+
+            # Auto-detect Grove availability and disable it if not present in the cluster.
+            # This allows tests to automatically adapt to cluster capabilities without manual configuration.
+            grove_available = await self.detect_grove_availability()
+            if not grove_available:
+                self._logger.info(
+                    "Grove not available in cluster, disabling Grove for deployment"
+                )
+                self.deployment_spec.disable_grove()
```

Comment:
do we need to explicitly disable grove or will the operator automatically handle it ?